ports/stm/supervisor/internal_flash.h

```diff
@@ -67,6 +68,12 @@
 #define INTERNAL_FLASH_FILESYSTEM_START_ADDR 0x08004000
 #endif
 
+#ifdef STM32F746xx
```

Comment:
I don't know why the ST hal doesn't differentiate between the STM32F746xE and the STM32F769xG variations the way it does for most chips with a flash variation. I'll revisit this later if necessary, but for now, could you add a warning stating that this applies to the F746xG only? The xE has only 512KB of flash memory.